bin/syclcc-clang

```diff
@@ -69,8 +69,7 @@ class config_file:
     if not key in self._data:
       return False
     if isinstance(self._data[key], str):
-      if(self._data[key].endswith("-NOTFOUND") or
-        self._data[key] == ""):
+      if(self._data[key].endswith("-NOTFOUND")):
```

Comment:
@illuhad I feel that this change could end up breaking a lot of stuff downstream, just consider it provisional: having an empty string as the value for a specific key in the configuration file could be considered correct to allow users explicitly set it to empty at configure time (e.g.: `-DCUDA_LINK_LINE=''`)?

Reply:
hm.. for CUDA and ROCm an empty string cannot be correct because you always need to link against some CUDA/ROCm libraries.. But I think I see how you arrived at this change for the CPU backend, where no additional linker arguments are necessary. I suspect the assumption that empty string means missing key may be quite important in order to detect missing settings in syclcc and also to make working with environment variables more comfortable (not requiring to explicitly having to unset them when they are not needed anymore). 
I wonder if a better course of action might not be to also move the `-lhipSYCL_cpu`, `-lhipSYCL_rocm` etc to the link line. In that case, the link line would never be empty and the user also has control over how the SYCL runtime is linked, if desired.

Reply:
Yep, it felt a bit too much to relax that check. I tried to think about your suggestion to move `-lhipSYCL_<platform>` flags out of `syclcc-clang` and into the config file but I didn't come up with a decent solution yet: given the fact that `-lhipSYCL_<platform>` flags make sense only when _not_ in bootstrap mode, having them in the config file (that right now contains only mandatory flags instead, valid both in and out of bootstrap mode) would require a pass to purge those known flags from linker args when in bootstrap mode. If you think that something like this could be acceptable I would go down this path tho.

Reply:
I tried to give a chance to the second solution:

* all hipSYCL linker flags are now pulled in from the config file instead of being hardcoded in `syclcc-clang`
* a separate config file to be used only during bootstrap is created at configure time (`syclcc-bootstrap.json`), this file contains no hipSYCL linker flags and unfortunately the empty string check (reintroduced) passes only because CMake leaves an empty space in the configuration template string (a bit sketchy imho, but works)
* the former `syclcc.json` is unused during bootstrap and contains all the hipSYCL linker flags needed by the deployed `syclcc-clang`